src/ZoneServer/Skills/Handlers/Swordsman/Hoplite/LongStride.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Melia.Shared.Data.Database;
+using Melia.Shared.Game.Const;
+using Melia.Shared.L10N;
+using Melia.Shared.World;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.Skills.Handlers.Base;
+using Melia.Zone.Skills.SplashAreas;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.Characters.Components;
+using Yggdrasil.Logging;
+using static Melia.Zone.Skills.SkillUseFunctions;
+
+namespace Melia.Zone.Skills.Handlers.Swordsman.Hoplite
+{
+	/// <summary>
+	/// Handler for the Hoplite skill Long Stride
+	/// </summary>
+	[SkillHandler(SkillId.Hoplite_LongStride)]
+	public class Hoplite_LongStride : IGroundSkillHandler, IDynamicCasted
+	{
+		/// <summary>
+		/// Called when the user starts casting the skill.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		public async void StartDynamicCast(Skill skill, ICombatEntity caster)
+		{
+		}
+
+		/// <summary>
+		/// Called when the user stops casting the skill.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		public void EndDynamicCast(Skill skill, ICombatEntity caster)
+		{
+		}
+
+		/// <summary>
+		/// Handles skill, damaging targets.
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="farPos"></param>
+		public void Handle(Skill skill, ICombatEntity caster, Position originPos, Position farPos, ICombatEntity target)
+		{
+			if (!skill.Vars.TryGet<Position>("Melia.ToolGroundPos", out var targetPos))
+			{
+				caster.ServerMessage(Localization.Get("No target location specified."));
+				return;
+			}
+
+			if (!caster.Map.Ground.IsValidPosition(targetPos))
+			{
+				caster.ServerMessage(Localization.Get("Invalid target location."));
+				return;
+			}
+			
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+			
+			skill.IncreaseOverheat();
+			caster.SetAttackState(true);
+
+			Send.ZC_SKILL_READY(caster, skill, originPos, targetPos);
+			Send.ZC_SKILL_MELEE_GROUND(caster, skill, targetPos, null);
```

Comment:
Random thought: Perhaps `ZC_SKILL_MELEE_GROUND` would be used for the sending of hits if there's no delay between them?

Reply:
I don't quite understand the comment, there definitely is a delay here.

Reply:
No, but there was an open question why it is used in some skills, and I was wondering whether these other skills might not have a delay, and then send their skill hits immediately with this packet instead of separately.